src/app/resolvers/user-list.resolver.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core'
```

Comment:
Why do we need a user-list.resolver? Why not call user.service straight from user-permission-management? There's a lot of layers atm and I'm just want to know why :)

Reply:
I'm not sure if it is necessarily a good practice, but right now I'm in favour of it :P

Specifying resolvers for routes allows us to resolve our data before we start loading the component. This means we do not need to have any logic for when no data is available/being fetched etc. I suppose this would have greater benefits when we are not dealing with lists of data, but rather an object's detailed view (as those are trickier to display without data).

Reply:
It's just four different parts to do one thing, I guess it will become clearer when we have some more data.